src/ast/passes/semantic_analyser.cpp

```diff
@@ -135,19 +135,19 @@ class SemanticAnalyser : public Visitor<SemanticAnalyser> {
   bool is_final_pass() const;
   bool is_first_pass() const;
 
-  bool check_assignment(const Call &call,
+  void check_assignment(const Call &call,
                         bool want_map,
                         bool want_var,
                         bool want_map_key);
+  [[nodiscard]] bool check_call(const Call &call);
   [[nodiscard]] bool check_nargs(const Call &call, size_t expected_nargs);
   [[nodiscard]] bool check_varargs(const Call &call,
                                    size_t min_nargs,
                                    size_t max_nargs);
   bool check_arg(const Call &call,
                  Type type,
                  int arg_num,
-                 bool want_literal = false,
-                 bool fail = true);
```

Comment:
There was only one call site using this last `fail` param. The site no longer calls this function.